src/ui/components/PromotedBadge/index.js

```diff
@@ -0,0 +1,105 @@
+/* @flow */
+import makeClassName from 'classnames';
+import * as React from 'react';
+import { compose } from 'redux';
+
+import translate from 'core/i18n/translate';
+import IconPromotedBadge from 'ui/components/IconPromotedBadge';
+import type {
+  PromotedBadgeCategory,
+  PromotedBadgeSize,
+} from 'ui/components/IconPromotedBadge';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+type Props = {|
+  category: PromotedBadgeCategory,
+  onClick?: Function | null,
+  size: PromotedBadgeSize,
+  sponsor?: boolean,
+|};
+
+export type InternalProps = {|
+  ...Props,
+  i18n: I18nType,
+|};
+
+export const PromotedBadgeBase = ({
+  category,
+  i18n,
+  onClick = null,
+  size,
+  sponsor = false,
+}: InternalProps) => {
+  let label;
+  let linkTitle;
+  let linkUrl;
+  switch (category) {
+    case 'line':
+      label = i18n.gettext('By Firefox');
+      // TODO: Update the title and URL when we know them.
```

Comment:
I wonder if we shouldn't add an l10n note to say not to translate these strings yet as it may be wasted work.

Reply:
That's a good point, thanks for raising it. I wonder if it would be best to just make them empty strings for now, which also wouldn't generate any translation needs. @jvillalobos and/or @willdurand WDYT?

Reply:
Either we remove the `gettext` calls and use English strings only or we keep things as is, it's not a lot of new strings and they are short.

Reply:
I don't like the idea of removing the `gettext` calls, because then I worry that someone might forget to put them back later. Let's just leave it as is.

Reply:
Holding off on translation for now is a good idea, yes. The content review for this stuff is still pending.